pvlib/iotools/solaranywhere.py

```diff
@@ -0,0 +1,285 @@
+    'Albedo_Unitless', 'DataVersion'
+]
+
+
+def get_solaranywhere(latitude, longitude, api_key, start=None, end=None,
+                      source='SolarAnywhereLatest', time_resolution=60,
+                      spatial_resolution=0.01, true_dynamics=False,
+                      probability_of_exceedance=None,
+                      variables=DEFAULT_VARIABLES, missing_data='FillAverage',
+                      url=URL, map_variables=True, max_response_time=300):
+    """Retrieve historical irradiance time series data from SolarAnywhere.
+
+    The SolarAnywhere API is described in [1]_ and [2]_. A detailed list of
+    available options for the input parameters can be found in [3]_.
+
+    Parameters
+    ----------
+    latitude: float
+        In decimal degrees, north is positive (ISO 19115).
+    longitude: float
+        In decimal degrees, east is positive (ISO 19115).
+    api_key: str
+        SolarAnywhere API key.
+    start: datetime like, optional
+        First timestamp of the requested period. If a timezone is not
+        specified, UTC is assumed. Not applicable for TMY data.
+    end: datetime like, optional
+        Last timtestamp of the requested period. If a timezone is not
+        specified, UTC is assumed. Not applicable for TMY data.
+    source: str, default: 'SolarAnywhereLatest'
+        Data source. Options include: 'SolarAnywhereLatest' (historical data),
+        'SolarAnywhereTGYLatest' (TMY for GHI), 'SolarAnywhereTDYLatest' (TMY
+        for DNI), or 'SolarAnywherePOELatest' for probability of exceedance.
+        Specific dataset versions can also be specified, e.g.,
+        'SolarAnywhere3_2' (see [3]_ for a full list of options).
+    time_resolution: {60, 30, 15, 5}, default: 60
+        Time resolution in minutes. For TMY data, time resolution has to be 60
+        min. (hourly).
+    spatial_resolution: {0.1, 0.01, 0.005}, default: 0.01
+        Spatial resolution in degrees.
+    true_dynamics: bool, default: False
+        Whether to apply SolarAnywhere TrueDynamics statistical processing.
+        Only available for the 5-min time resolution.
+    probability_of_exceedance: int, optional
+        Probability of exceedance in the range of 1 to 99. Only relevant when
+        requesting probability of exceedance (POE) time series.
+    variables: list-like, default: :const:`DEFAULT_VARIABLES`
+        Variables to retrieve (described in [4]_).  Available variables depend
+        on whether historical or TMY data is requested.
+    missing_data: {'Omit', 'FillAverage'}, default: 'FillAverage'
+        Method for treating missing data.
+    url: str, default: :const:`pvlib.iotools.solaranywhere.URL`
+        Base url of SolarAnywhere API.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    max_response_time: float, default: 300
```

Comment:
Some other iotools functions have a `timeout` parameter that serves a similar purpose (although the mechanical details are a bit different under the hood).  Worth considering using the same parameter name here for consistency.

Reply:
Has now been implemented